airbyte-integrations/connectors/source-mixpanel/source_mixpanel/source.py

```diff
@@ -0,0 +1,612 @@
+from datetime import date, timedelta
+from pprint import pprint
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union
+from urllib.parse import parse_qs, urlparse
+
+import requests
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator, TokenAuthenticator
+
+
+class MixpanelStream(HttpStream, ABC):
+
+    url_base = "https://mixpanel.com/api/2.0/"
+    date_window_size = 30  # days
+
+    # https://help.mixpanel.com/hc/en-us/articles/115004602563-Rate-Limits-for-Export-API-Endpoints#api-export-endpoint-rate-limits
+    rate_limit = 400  # 400 queries per hour == 1 req in 9 secs
+
+    def __init__(
+        self,
+        authenticator: HttpAuthenticator,
+        start_date: Union[date, str] = None,
+        end_date: Union[date, str] = None,
+        date_window_size: int = 30,  # in days
+        **kwargs,
+    ):
+        self.start_date = start_date
+        self.end_date = end_date
+        self.date_window_size = date_window_size
+
+        super().__init__(authenticator=authenticator)
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def date_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        date_slices = []
+
+        # use the latest date between self.start_date and stream_state
+        start_date = max(self.start_date, date.fromisoformat(stream_state["date"]) if stream_state else self.start_date)
+        # use the lowest date between start_date and self.end_date, otherwise API fails if start_date is in future
+        start_date = min(start_date, self.end_date)
+
+        while start_date <= self.end_date:
+            end_date = start_date + timedelta(days=self.date_window_size - 1)  # -1 is needed because dates are inclusive
+            date_slices.append(
+                {
+                    "start_date": str(start_date),
+                    "end_date": str(min(end_date, self.end_date)),
+                }
+            )
+            # add 1 additional day because date range is inclusive
+            start_date = end_date + timedelta(days=1)
+
+        print(f"==== date_slices len: {len(date_slices)} \n")
+        pprint(date_slices)
+        return date_slices
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {"Accept": "application/json"}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {}
```

Comment:
Parent HttpStream already returns {}. I think no need in this request_params method here